quick_start/kusto_sample_app.py

```diff
@@ -0,0 +1,261 @@
+    time.sleep(20)
+
+    print("")
+    print(f"Post ingestion row count for '{databaseName}.{tableName}' is:")
+    run_query(kusto_client, databaseName, f"{tableName} | summarize count()")
+
+
+def run_control_command(client: KustoClient, db: str, command: str) -> bool:
+    try:
+        res = client.execute_mgmt(db, command)
+        for row in res.primary_results[0]:
+            print(row.to_list())
+
+        return True
+
+    except KustoClientError as ex:
+        die(f"Client error while trying to execute command '{command}' on database '{db}'", ex)
+
+    except KustoServiceError as ex:
+        die(f"Server error while trying to execute command '{command}' on database '{db}'", ex)
+
+    except Exception as ex:
+        die(f"Unknown error while trying to execute command '{command}' on database '{db}'", ex)
+
+    return False
+
+
+def run_query(client: KustoClient, db: str, query: str):
+    try:
+        res = client.execute_query(db, query)
+        for row in res.primary_results[0]:
+            print(row.to_list())
+
+        return True
+
+    except KustoClientError as ex:
+        die(f"Client error while trying to execute query '{query}' on database '{db}'", ex)
+
+    except KustoServiceError as ex:
+        die(f"Server error while trying to execute query '{query}' on database '{db}'", ex)
+
+    except Exception as ex:
+        die(f"Unknown error while trying to execute query '{query}' on database '{db}'", ex)
+
+    return False
+
+
+def ingest_data_from_file(client: QueuedIngestClient, db: str, table: str, file_path: str):
+    ingestion_props = IngestionProperties(
+        database=f"{databaseName}",
+        table=f"{tableName}",
+        data_format=fileFormat,
+
+        # Todo - Configure: Setting the ingestion batching policy takes up to 5 minutes to have an effect.
+        #  For the sake of the sample we set Flush-Immediately, but in practice it should not be commonly used.
+        #  Comment the below line after running the sample for the first few times!
+        flush_immediately=True,
+
+        # Todo - Tip: in case a mapping is required provide either a mapping or a mapping reference for a pre-configured mapping
+        # ingestion_mapping_type=IngestionMappingType.JSON
+        # ingestion_mapping=
+        # ingestion_mapping_reference="{json_mapping_that_already_exists_on_table}"
+    )
+
+    # Todo - Tip: for optimal ingestion batching it's best to specify the uncompressed data size in the file descriptor
+    file_descriptor = FileDescriptor(f"{file_path}")
+    client.ingest_from_file(file_descriptor, ingestion_properties=ingestion_props)
+
+    # Todo - Tip: Kusto can also ingest data from blobs, open streams and pandas dataframes.
+    #  See the python SDK azure.kusto.ingest samples for additional references.
+
+
+def create_connection_string(cluster: str, auth_mode: str) -> KustoConnectionStringBuilder:
+    if auth_mode == "deviceCode":

```

Comment:
this kinda works against our static functions approach. if a user were to copy-paste this...

Reply:
Explain?

Reply:
`KustoConnectionStringBuilder.with_method` should be clear and explicit enough. have a wrapper function might steer users to doing the same, which would make our static function obsolete. It is not a bad thing per se, just a comment.

Reply:
Oh, that's just cause the script demonstrates multiple auth modes (certificates, keys, etc.).
You need some way of selecting between them and I did not want to push this code into the main function.